fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -1011,7 +1028,21 @@ public Type getType() {
         @Nullable
         @Override
         public Boolean eval(@Nonnull FDBRecordStoreBase<?> store, @Nonnull EvaluationContext context, @Nullable Object value) {
-            return evalListComparison(type, value, comparand);
+            if (value == null) {
+                return null;
+            }
+            switch (type) {
+                case EQUALS:
+                    return value.equals(comparand);
+                case NOT_EQUALS:
+                    return !value.equals(comparand);
+                case STARTS_WITH:
+                    return compareListStartsWith(value, comparand);
```

Comment:
It feels kind of wrong that this duplicates the logic in `evalListComparison`. It seems like this should instead check if it's an `IN` (and perhaps validate that `inProbe` is not `null`), and then otherwise fall back to `evalListComparison`.

Reply:
I like that, will change.  Thanks Alec.